Notebooks/OOPS_TMT_Statistical_testing.Rmd

```diff
@@ -123,52 +135,73 @@ volcano_plot <- function(lmResultsObj,
 ```
 
 ```{r}
-
-lmResults <- run_limma(data_to_model, design, coefficient='materialRBP:toxicToxic')
+limma_fit <- run_limma(data_to_model, design)
+lmResults <- summariseLmFit(limma_fit, coefficient='materialRBP:toxicToxic')
 
 volcano <- volcano_plot(lmResults,
                           adj.P.Val_threshold=0.1,
                           logFC_threshold=0.9,
                           xlabel='Effect of Toxic ASO on RNA binding (log2)')
 
 #define function build_results_list
-build_results_list <- function(data_to_model, design, results, volcano){
+build_results_list <- function(data_to_model, design, limma_fit, results, volcano){
   list('data'=data_to_model,
        'design'=design,
+       'limma_fit'=limma_fit,
        'results'=results,
        'volcano'=volcano)
 }
 
-lm_results <- build_results_list(data_to_model,
-                                   design,
-                                   lmResults,
-                                   volcano)
-print(lm_results$volcano)
+lm_results_toxic_vs_non_toxic <- build_results_list(data_to_model,
+                                 design,
+                                 limma_fit,
+                                 lmResults,
+                                 volcano)
+
+print(lm_results_toxic_vs_non_toxic$volcano)
 
 
 ```
 
 
 
 ```{r}
-library(biobroom)
-p<- tidy(data_to_model, addPheno=TRUE)
 
-q<- p %>% filter(protein=="Q15050")
 
-q$value <- log2(q$value)
+# lmResults is ordered by p-value, so first protein has smallest p-value.
```

Comment:
Generalises the code so that it identifies the protein with the lowest p-value and plots that. No manual specification of the protein ID